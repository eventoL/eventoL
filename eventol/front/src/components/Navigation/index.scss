@import '../../styles/index';

.navigation {
  nav {
    ul {
      button {
        background-color: transparent;
        border: 0;
        color: $white;
        display: contents;
        font-size: 100%;
        margin: 0;
        padding: 0;
        vertical-align: baseline;
      }
    }
  }

  .languages-dropdown {
    background-color: transparent;
    border-radius: 3px;
    color: $white;
    display: list-item;
    font-size: 14px;
    font-weight: 400;
    padding: 7px 10px;
    vertical-align: baseline;

    .dropbtn {
      i {
        margin-left: 5px;
      }
    }

    .dropdown-content {
      background-color: $dropdown-background-color;
      box-shadow: 0 8px 16px 0 $dropdown-box-shadow-color;
      display: grid;
      margin-top: 5px;
      min-width: 160px;
      position: relative;
      z-index: 1;

      a {
        border-radius: 3px;
        display: block;
        float: none;
        font-size: 14px;
        font-weight: 400;
        padding: 7px 10px;
        text-align: left;
        text-decoration: none;
        transition: background .125s ease;

        &:hover {
          background: $dropdown-background-hover-color;
          cursor: pointer;
        }
      }
    }
  }
}

@media (min-width: 950px) {
  .navigation {
    .languages-dropdown {
      &:hover {
        .dropdown-content {
          display: block;
          position: fixed;
        }
      }

      .dropdown-content {
        display: none;
        position: unset;
      }
    }

    ul {
      @include flex-element;
      -ms-flex-align: center;
      -webkit-align-items: center;
      -webkit-box-align: center;
      align-items: center;
      height: 37px;
      padding: 0 10px;

      li {
        -webkit-transition: background .125s ease;
        border-radius: 3px;
        font-size: 14px;
        font-weight: 400;
        padding: 7px 10px;
        transition: background .125s ease;

        a {
          text-decoration: none;
        }
      }

      a {
        text-decoration: none;
      }

      li:hover {
        background: $eventol-color;
        cursor: pointer;
      }
    }
  }
}

@media (max-width: 950px) {
  @mixin li-element {
    -webkit-transition: background .125s ease;
    border-radius: 3px;
    font-size: 14px;
    font-weight: 400;
    padding: 7px 10px;
    text-align: left;
    transition: background .125s ease;

    &:hover {
      background: $eventol-color;
      cursor: pointer;
    }
  }

  .navigation {
    ul {
      background: $gray;
      margin-right: 8vw;
      opacity: .8;
      z-index: 0;

      a {
        // sass-lint:disable no-important
        float: unset !important;
        // sass-lint:enable no-important
        text-decoration: none;

        li {
          @include li-element;
        }
      }

      li {
        @include li-element;
      }
    }
  }
}
