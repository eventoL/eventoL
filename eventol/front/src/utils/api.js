import _ from 'lodash';
import Cookies from 'js-cookie';

import Logger from './logger';
import {getReportUrl} from './urls';

export const getCsrf = () => Cookies.get('csrftoken');

export const addQueryString = (url, query) => {
  const queryKeys = query ? Object.keys(query) : [];
  if (!_.isEmpty(queryKeys)) {
    const params = queryKeys.map(k => `${k}=${encodeURIComponent(query[k])}`);
    return `${url}?${params.join('&')}`;
  }
  return url;
};

export const genericFetch = (url, queryString, params) => {
  const newUrl = addQueryString(url, queryString);
  return fetch(newUrl, params)
    .then(res => {
      if (res.ok) {
        return res.json();
      }
      return Promise.reject(res);
    })
    .catch(res => {
      if (res.status === 403) {
        window.location.hash = '/';
        window.location.href = '/';
        Logger.warning('Redirect to /', res.status, res.statusText);
      } else {
        Logger.error('There has been an error', res.status, res.statusText);
      }
    });
};

export const getUrl = (url, queryString) =>
  genericFetch(url, queryString, {
    method: 'GET',
    credentials: 'same-origin',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      'X-CSRFToken': getCsrf(),
    },
  });

export const postUrl = (url, data, queryString) =>
  genericFetch(url, queryString, {
    method: 'POST',
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json',
      'X-CSRFToken': getCsrf(),
    },
    body: JSON.stringify(data),
    credentials: 'same-origin',
  });

export const loadReports = (pageSize, page, sorted) =>
  getUrl(getReportUrl(pageSize, page, sorted));
