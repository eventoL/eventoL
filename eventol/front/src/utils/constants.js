export const MOBILE_WIDTH = 950;
export const LOGO_LANDING_DEFAULT = '/static/manager/img/logo.png';
export const BACKGROUND_DEFAULT = '/static/manager/img/background.png';
export const LOGO_HEADER_DEFAULT = '/static/manager/img/eventol-white.png';
export const REPORT_REQUIRED_FIELDS =
  'name,event_slug,email,location,report,id,tags';
export const HOME_REQUIRED_FIELDS =
  'event_slug,place,image,name,attendees_count,abstract,tags';
export const MAP_LAYER =
  'https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png';

export const SLIDER_BASE_SETTINGS = {
  variableWidth: false,
  dots: true,
  infinite: false,
  speed: 500,
  slidesToShow: 4,
  slidesToScroll: 4,
  initialSlide: 0,
  responsive: [
    {
      breakpoint: 1290,
      settings: {
        slidesToShow: 3,
        slidesToScroll: 3,
      },
    },
    {
      breakpoint: 900,
      settings: {
        slidesToShow: 2,
        slidesToScroll: 2,
        initialSlide: 2,
      },
    },
    {
      breakpoint: 610,
      settings: {
        slidesToShow: 1,
        slidesToScroll: 1,
      },
    },
  ],
};

export const MAP_SETTINGS = {
  scrollWheelZoom: false,
  zoomControl: false,
  dragging: false,
  keyboard: false,
  doubleClickZoom: false,
  touchZoom: false,
  boxZoom: false,
  minZoom: 11,
  maxZoom: 11,
};
