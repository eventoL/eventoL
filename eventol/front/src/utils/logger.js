/* eslint-disable no-console */

const Logger = {
  error: console.error,
  warning: console.warn,
  log: console.log,
};

export default Logger;
