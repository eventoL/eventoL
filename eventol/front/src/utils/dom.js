import {MOBILE_WIDTH} from './constants';

export const focusOn = id => {
  document.getElementById(id).focus();
};

export const mapSizesToProps = ({width}) => ({
  isMobile: width < MOBILE_WIDTH,
});
