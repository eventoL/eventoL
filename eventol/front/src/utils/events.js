import {getEventUrl} from './urls';
import {LOGO_LANDING_DEFAULT} from './constants';


export const getEvent = (pk, fields) => ({pk, fields});
export const getSlugParsed = slug => slug
  .toLowerCase()
  .split('-')
  .join(' ')
  .split('_')
  .join(' ')
  .split(' ')
  .map(word => ((word) ? word.replace(word[0], word[0].toUpperCase()) : ''))
  .join(' ');

export const parseEventToItem = ({
  tags, place, image,
  name: title, attendees_count: attendees,
  abstract: overview, event_slug: eventSlug,
}) => {
  let backdrop = image;
  if (image){
    backdrop = new URL(image).pathname;
  }
  return {
    eventSlug,
    title,
    attendees,
    overview,
    backdrop,
    place,
    tags,
    key: eventSlug,
    url: getEventUrl(eventSlug),
  };
};

export const emptyEventItem = {
  key: 'not_found',
  title: gettext('Event not found'),
  overview: gettext('No Event found in your search'),
  backdrop: LOGO_LANDING_DEFAULT,
};
