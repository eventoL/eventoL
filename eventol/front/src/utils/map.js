import Logger from './logger';
import {MAP_SETTINGS, MAP_LAYER} from './constants';

export const getMapId = (eventSlug, sliderId) => `${sliderId}${eventSlug}`;

export const loadMap = (place, mapId) => {
  try {
    const {geometry} = JSON.parse(place);
    const eventLat = geometry.location.lat;
    const eventLon = geometry.location.lng;
    const map = L.map(mapId, MAP_SETTINGS).setView([eventLat, eventLon], 5);
    L.tileLayer(MAP_LAYER, {attribution: ''}).addTo(map);
    map.attributionControl.setPrefix('');
    L.marker([eventLat, eventLon]).addTo(map);
  } catch (e) {
    Logger.error(e);
  }
};
