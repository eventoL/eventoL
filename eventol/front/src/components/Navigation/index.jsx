import React from 'react';
import PropTypes from 'prop-types';

import {focusOn} from '../../utils/dom';
import {EVENTOL_DOCUMENTATION, REPORT_URL} from '../../utils/urls';

import './index.scss';

export default class Navigation extends React.PureComponent {
  static propTypes = {
    handlerOnChangeLanguage: PropTypes.func.isRequired,
    languages: PropTypes.arrayOf(
      PropTypes.shape({
        code: PropTypes.string,
        name: PropTypes.string,
      })
    ),
  };

  static defaultProps = {
    languages: [],
  };

  handleSearchFocus = event => {
    event.preventDefault();
    event.stopPropagation();
    focusOn('search-input');
  };

  render() {
    const {languages, handlerOnChangeLanguage} = this.props;
    return (
      <div className="navigation" id="navigation">
        <nav>
          <ul>
            <button
              onClick={this.handleSearchFocus}
              onKeyPress={this.handleSearchFocus}
              role="link"
              tabIndex="0"
              type="button"
            >
              <li>{gettext('Search')}</li>
            </button>
            <a href={EVENTOL_DOCUMENTATION}>
              <li>{gettext('Documentation')}</li>
            </a>
            <a href="#featured">
              <li>{gettext('Featured')}</li>
            </a>
            <a href="#recent">
              <li>{gettext('Recent')}</li>
            </a>
            <a href={REPORT_URL}>
              <li>{gettext('Generic report')}</li>
            </a>
            {languages.length > 0 && (
              <div className="languages-dropdown">
                <button className="dropbtn" type="button">
                  {gettext('Languages')}
                  <i className="fa fa-caret-down" />
                </button>
                <div className="dropdown-content">
                  {languages.map(({code, name}) => (
                    <a
                      key={code}
                      href="#lang"
                      id={code}
                      onClick={() => handlerOnChangeLanguage(code)}
                    >
                      {name}
                    </a>
                  ))}
                </div>
              </div>
            )}
          </ul>
        </nav>
      </div>
    );
  }
}
