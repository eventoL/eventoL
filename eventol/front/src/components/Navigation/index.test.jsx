import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../../utils/dom', () => ({
  focusOn: jest.fn(),
}));

jest.mock('../../utils/urls', () => ({
  EVENTOL_DOCUMENTATION: 'EVENTOL_DOCUMENTATION',
  REPORT_URL: 'REPORT_URL',
}));

import Navigation from '.';
import {focusOn} from '../../utils/dom';

describe('Navigation', () => {
  let tree;
  let event;
  let instance;
  let component;
  let languages;
  const handlerOnChangeLanguage = jest.fn();

  const getComponent = props => {
    component = renderer.create(<Navigation {...props} />);
    instance = component.root;
    return component;
  };

  beforeEach(() => {
    event = {
      preventDefault: jest.fn(),
      stopPropagation: jest.fn(),
    };
    languages = [
      {code: 'es', name: 'Spanish'},
      {code: 'en', name: 'English'},
    ];
    component = getComponent();
    tree = component.toJSON();
  });

  afterEach(() => {
    jest.clearAllMocks();
    focusOn.mockClear();
  });

  test('should call event.stopPropagation, event.preventDefault and focusOn on click event', () => {
    expect(event.preventDefault).not.toBeCalled();
    expect(event.stopPropagation).not.toBeCalled();
    expect(focusOn).not.toBeCalled();

    instance.findByType('button').props.onClick(event);

    expect(event.preventDefault).toBeCalled();
    expect(event.stopPropagation).toBeCalled();
    expect(focusOn).toBeCalled();
  });

  test('should call event.stopPropagation, event.preventDefault and focusOn on key press event', () => {
    expect(event.preventDefault).not.toBeCalled();
    expect(event.stopPropagation).not.toBeCalled();
    expect(focusOn).not.toBeCalled();

    instance.findByType('button').props.onKeyPress(event);

    expect(event.preventDefault).toBeCalled();
    expect(event.stopPropagation).toBeCalled();
    expect(focusOn).toBeCalled();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('with languages', () => {
    beforeEach(() => {
      component = getComponent({languages, handlerOnChangeLanguage});
      tree = component.toJSON();
    });

    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });

    test('Dropdown, options and on click handlers (handlerOnChangeLanguage)', () => {
      const languagesDropdown = instance.findByProps({
        className: 'languages-dropdown',
      });

      expect(languagesDropdown).toBeDefined();

      const options = languagesDropdown.findAllByType('a');
      expect(options).toBeDefined();
      expect(options).toHaveLength(2);

      languages.forEach(({code, name}) => {
        const option = options.find(op => op.props.id === code);
        expect(option).toBeDefined();
        expect(option.children[0]).toEqual(name);

        // on Click (handlerOnChangeLanguage)
        handlerOnChangeLanguage.mockClear();

        option.props.onClick();
        expect(handlerOnChangeLanguage).toBeCalled();
        expect(handlerOnChangeLanguage).toBeCalledWith(code);
      });
    });
  });
});
