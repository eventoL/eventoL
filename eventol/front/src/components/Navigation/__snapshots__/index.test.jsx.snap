// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Default Snapshot 1`] = `
<div
  className="navigation"
  id="navigation"
>
  <nav>
    <ul>
      <button
        onClick={[Function]}
        onKeyPress={[Function]}
        role="link"
        tabIndex="0"
        type="button"
      >
        <li>
          Search
        </li>
      </button>
      <a
        href="EVENTOL_DOCUMENTATION"
      >
        <li>
          Documentation
        </li>
      </a>
      <a
        href="#featured"
      >
        <li>
          Featured
        </li>
      </a>
      <a
        href="#recent"
      >
        <li>
          Recent
        </li>
      </a>
      <a
        href="REPORT_URL"
      >
        <li>
          Generic report
        </li>
      </a>
    </ul>
  </nav>
</div>
`;

exports[`Navigation with languages Snapshot 1`] = `
<div
  className="navigation"
  id="navigation"
>
  <nav>
    <ul>
      <button
        onClick={[Function]}
        onKeyPress={[Function]}
        role="link"
        tabIndex="0"
        type="button"
      >
        <li>
          Search
        </li>
      </button>
      <a
        href="EVENTOL_DOCUMENTATION"
      >
        <li>
          Documentation
        </li>
      </a>
      <a
        href="#featured"
      >
        <li>
          Featured
        </li>
      </a>
      <a
        href="#recent"
      >
        <li>
          Recent
        </li>
      </a>
      <a
        href="REPORT_URL"
      >
        <li>
          Generic report
        </li>
      </a>
      <div
        className="languages-dropdown"
      >
        <button
          className="dropbtn"
          type="button"
        >
          Languages
          <i
            className="fa fa-caret-down"
          />
        </button>
        <div
          className="dropdown-content"
        >
          <a
            href="#lang"
            id="es"
            onClick={[Function]}
          >
            Spanish
          </a>
          <a
            href="#lang"
            id="en"
            onClick={[Function]}
          >
            English
          </a>
        </div>
      </div>
    </ul>
  </nav>
</div>
`;
